build-support/bin/ci.py

```diff
@@ -0,0 +1,418 @@
+#!/usr/bin/env python3
+# Copyright 2019 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+import argparse
+import os
+import platform
+import subprocess
+from enum import Enum
+from pathlib import Path
+from typing import List, Optional
+
+from common import banner, die, green, travis_section
+
+
+def main() -> None:
+  banner("CI BEGINS")
+
+  args = create_parser().parse_args()
+  setup_environment(python_version=args.python_version)
+
+  if args.bootstrap:
+    bootstrap(clean=args.bootstrap_clean, python_version=args.python_version)
+  set_run_from_pex()
+
+  if args.precommit_checks:
+    run_precommit_checks()
+  if args.sanity_checks:
+    run_sanity_checks()
+  if args.lint:
+    run_lint()
+  if args.doc_gen:
+    run_doc_gen_tests()
+  if args.clippy:
+    run_clippy()
+  if args.cargo_audit:
+    run_cargo_audit()
+  if args.python_tests:
+    run_python_tests()
+  if args.rust_tests:
+    run_rust_tests()
+  if args.jvm_tests:
+    run_jvm_tests()
+  if args.integration_tests:
+    run_integration_tests(shard=args.integration_shard)
+  if args.internal_backend_tests:
+    run_internal_backend_tests()
+  if args.platform_specific_tests:
+    run_platform_specific_tests()
+
+  banner("CI ENDS")
+  print()
+  green("SUCCESS")
+
+# -------------------------------------------------------------------------
+# Options
+# -------------------------------------------------------------------------
+
+class PythonVersion(Enum):
+  py27 = "2.7"
+  py36 = "3.6"
+  py37 = "3.7"
+
+  def __str__(self) -> str:
+    return str(self.value)
+
+
+def create_parser() -> argparse.ArgumentParser:
+  # TODO: reassign the short CLI flag names to reflect the long names.
+  parser = argparse.ArgumentParser(description="Runs commons tests for local or hosted CI.")
+  parser.add_argument(
+    "-v", "--python-version",
+    type=PythonVersion,
+    choices=list(PythonVersion),
+    default=PythonVersion.py36,
+    help="Run Pants with this version (defaults to 3.6)."
+  )
+  parser.add_argument(
```

Comment:
As a follow-up, this should probably die. Any test command using:
```python
_run_command(
  ["./pants.pex", ...
  ...
)
```
has an implicit dependency on `--bootstrap` being present on the CLI to succeed.